chainercv/links/model/pspnet/pspnet.py

```diff
@@ -0,0 +1,476 @@
+                    out_ch, comm, eps=1e-5, decay=0.95)
+            else:
+                self.bn = L.BatchNormalization(out_ch, eps=1e-5, decay=0.95)
+
+    def __call__(self, x, relu=True):
+        h = self.bn(self.conv(x))
+        return h if not relu else F.relu(h)
+
+
+class PyramidPoolingModule(chainer.ChainList):
+
+    def __init__(self, in_ch, feat_size, pyramids):
+        super(PyramidPoolingModule, self).__init__(
+            ConvBNReLU(in_ch, in_ch // len(pyramids), 1, 1, 0),
+            ConvBNReLU(in_ch, in_ch // len(pyramids), 1, 1, 0),
+            ConvBNReLU(in_ch, in_ch // len(pyramids), 1, 1, 0),
+            ConvBNReLU(in_ch, in_ch // len(pyramids), 1, 1, 0))
+        if isinstance(feat_size, int):
+            self.ksizes = (feat_size // np.array(pyramids)).tolist()
+        elif isinstance(feat_size, (list, tuple)) and len(feat_size) == 2:
+            kh = (feat_size[0] // np.array(pyramids)).tolist()
+            kw = (feat_size[1] // np.array(pyramids)).tolist()
+            self.ksizes = list(zip(kh, kw))
+
+    def __call__(self, x):
+        ys = [x]
+        h, w = x.shape[2:]
+        for f, ksize in zip(self, self.ksizes):
+            y = F.average_pooling_2d(x, ksize, ksize)  # Pad should be 0!
+            y = f(y)  # Reduce num of channels
+            y = F.resize_images(y, (h, w))
+            ys.append(y)
+        return F.concat(ys, axis=1)
+
+
+class BottleneckConv(chainer.Chain):
+
+    def __init__(self, in_ch, mid_ch, out_ch, stride=2, dilate=False):
+        mid_stride = chainer.config.mid_stride
+        super(BottleneckConv, self).__init__()
+        with self.init_scope():
+            self.cbr1 = ConvBNReLU(
+                in_ch, mid_ch, 1, 1 if mid_stride else stride, 0)
+            if dilate:
+                self.cbr2 = ConvBNReLU(mid_ch, mid_ch, 3, 1, dilate, dilate)
+            else:
+                self.cbr2 = ConvBNReLU(
+                    mid_ch, mid_ch, 3, stride if mid_stride else 1, 1)
+            self.cbr3 = ConvBNReLU(mid_ch, out_ch, 1, 1, 0)
+            self.cbr4 = ConvBNReLU(in_ch, out_ch, 1, stride, 0)
+
+    def __call__(self, x):
+        h = self.cbr1(x)
+        h = self.cbr2(h)
+        h1 = self.cbr3(h, relu=False)
+        h2 = self.cbr4(x, relu=False)
+        return F.relu(h1 + h2)
+
+
+class BottleneckIdentity(chainer.Chain):
+
+    def __init__(self, in_ch, mid_ch, dilate=False):
+        super(BottleneckIdentity, self).__init__()
+        with self.init_scope():
+            self.cbr1 = ConvBNReLU(in_ch, mid_ch, 1, 1, 0)
+            if dilate:
+                self.cbr2 = ConvBNReLU(mid_ch, mid_ch, 3, 1, dilate, dilate)
+            else:
+                self.cbr2 = ConvBNReLU(mid_ch, mid_ch, 3, 1, 1)
+            self.cbr3 = ConvBNReLU(mid_ch, in_ch, 1, 1, 0)
+
+    def __call__(self, x):
+        h = self.cbr1(x)
+        h = self.cbr2(h)
+        h = self.cbr3(h, relu=False)
+        return F.relu(h + x)
+
+
+class ResBlock(chainer.ChainList):
```

Comment:
Can you merge `ResBlock` and `DilatedResBlock`?

Reply:
Merged.